src/main/java/it/pagopa/interop/probing/scheduler/scheduler/ScheduledTasks.java

```diff
@@ -32,27 +33,35 @@ public class ScheduledTasks {
   @Value("${scheduler.limit}")
   private Integer limit;
 
+  private Integer offset;
+
 
   @Scheduled(cron = "${scheduler.cron.expression}")
   public void scheduleFixedDelayTask() {
     MDC.put(LoggingPlaceholders.TRACE_ID_PLACEHOLDER,
         "- [CID= " + UUID.randomUUID().toString().toLowerCase() + "]");
     logger.logSchedulerStart();
     try {
-      Integer offset = 0;
+      offset = 0;
       while (true) {
-        PollingEserviceResponse response =
-            eserviceService.getEservicesReadyForPolling(limit, offset);
-        for (EserviceContent service : response.getContent()) {
+        CompletableFuture<PollingEserviceResponse> response = CompletableFuture
```

Comment:
It's all right , but it is better to create an proper ExecutionContext for non blocking operations such as newFixedThreadPool.
At this proposite please get in touch with your colleague that are using Spring @Async, it is the same thing but for readability it is better to use the same  pattern.